src/bin/nydus-image/merge.rs

```diff
@@ -72,15 +72,15 @@ impl Merger {
             let rs = RafsSuper::load_from_metadata(bootstrap_path, RafsMode::Direct, true)
                 .context(format!("load bootstrap {:?}", bootstrap_path))?;
 
-            if let Some(config) = &config {
-                config.check_compatibility(&rs.meta)?;
-            } else {
+            if config.is_none() {
```

Comment:
How about this?
```
let config = config.as_ref().unwrap_or_else(|| rs.meta.get_config());
```

Reply:
Thanks for the comment, fixed by using `get_or_insert_with`.